xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -4968,9 +4970,19 @@ private void updateLinksForRename(XWikiDocument sourceDoc, DocumentReference new
         List<DocumentReference> backlinkDocumentReferences, List<DocumentReference> childDocumentReferences,
         XWikiContext context) throws XWikiException
     {
+        // FIXME: that's ugly we should use something else.
+        JobContext jobContext = Utils.getComponent(JobContext.class);
+        Job currentJob = jobContext.getCurrentJob();
+
+        Map<EntityReference, EntityReference> updatedReferences =
+            Map.of(sourceDoc.getDocumentReference(), newDocumentReference);
+        if (currentJob instanceof AbstractCopyOrMoveJob) {
+            updatedReferences = ((AbstractCopyOrMoveJob) currentJob).getSelectedEntities();
+        }
```

Comment:
@tmortagne for that one you suggested: 
> But I'm not sure it's a good thing to do, might be cleaner to introduce something more genereric in the ExecutionContext that would be populated by jobs like the move job, etc.

wdyt, the PR is already quite complex, isn't it something we can address later?

Reply:
Sure, it's not a blocker, just a cleaner thing to do eventually for the long run. As long as we don't introduce public API for the current version, it's not adding too much debt.